httpserver/serialization.cpp

```diff
@@ -77,7 +77,7 @@ static void serialize_uint64(std::string& buf, uint64_t a) {
 
 static void serialize(std::string& buf, const std::string& str) {
 
-    buf.reserve(buf.size() + str.size() + 4);
+    buf.reserve(buf.capacity() + str.size() + sizeof(uint32_t));
```

Comment:
Why do you think capacity should be used here? The intention was to make sure there is enough room in the buffer, not to increase the buffer unconditionally. However, I'm not wondering if `reserve` is going to help much, since we only push two times there.

Reply:
if `bug.size() + str.size() + 4` happens to be smaller than your capacity, you basically allow the container to shrink, no?

Reply:
Ok re-reading your comment made me change my mind.

Reply:
If it is smaller 
> the function call does not cause a reallocation and the vector capacity is not affected

Reply:
Well I guess we could still merge that `sizeof(uint32_t)` change ;)